bitcoin/BRSyncManager.c

```diff
@@ -1279,7 +1279,9 @@ BRClientSyncManagerAnnounceGetTransactionsDone (BRClientSyncManager manager,
                                                       (const char **) addresses,
                                                       addressCount,
                                                       begBlockNumber,
-                                                      endBlockNumber,
+                                                      (BRClientSyncManagerScanStateIsFullScan (&manager->scanState)
```

Comment:
Can this behaviour be folded into `BRClientSyncManagerScanStateGetEndBlockNumber`? The idea with all those `BRClientSyncManagerScanState` functions was to encapsulate, as much as possible, the scan behaviour/logic.

Reply:
(alternatively, could set the `scanState->endBlockNumber` in `BRClientSyncManagerScanStateInit`, allowing `BRClientSyncManagerScanStateGetEndBlockNumber` to remain the same.

Reply:
If so, please go ahead and change as you see fit.

Reply:
Okay, give me 10.